runtime/j9vm/java9vmi.c

```diff
@@ -1697,23 +1698,18 @@ JVM_GetModuleByPackageName(JNIEnv *env, jobject classLoader, jstring packageName
 			thisVMClassLoader = J9VMJAVALANGCLASSLOADER_VMREF(currentThread, thisClassloader);
 		}
 
-		length = vmFuncs->getStringUTF8Length(currentThread, packageObj);
-		packageUTF8 = (J9UTF8 *) buf;
+		packageName = vmFuncs->copyStringToUTF8WithMemAlloc(currentThread, packageObj, J9_STR_NONE, "", 0, buf, J9VM_PACKAGE_NAME_BUFFER_LENGTH, &length);
 
-		if (length > J9VM_PACKAGE_NAME_BUFFER_LENGTH) {
-			packageUTF8 = j9mem_allocate_memory(length + sizeof(packageUTF8->length) + 1, OMRMEM_CATEGORY_VM);
-			if (NULL == packageUTF8) {
-				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-				goto exit;
-			}
+		if (NULL == packageName) {
+			vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto exit;
 		}
-		vmFuncs->copyStringToUTF8Helper(currentThread, packageObj, TRUE, J9_STR_NONE, J9UTF8_DATA(packageUTF8), length + 1);
-		J9UTF8_SET_LENGTH(packageUTF8, (U_16) length);
-		if (NULL != strchr((const char*)J9UTF8_DATA(packageUTF8), '.')) {
+		
+		if (NULL != strchr((const char*)packageName, '.')) {
 			vmFuncs->setCurrentExceptionUTF(currentThread, J9VMCONSTANTPOOL_JAVALANGILLEGALARGUMENTEXCEPTION, "package name contains '.' instead of '/'");
 		} else {
-			if (vmFuncs->isAnyClassLoadedFromPackage(thisVMClassLoader, J9UTF8_DATA(packageUTF8), J9UTF8_LENGTH(packageUTF8))) {
-				vmModule = vmFuncs->findModuleForPackageUTF8(currentThread, thisVMClassLoader, packageUTF8);
+			if (vmFuncs->isAnyClassLoadedFromPackage(thisVMClassLoader, packageName, packageLength)) {
+				vmModule = vmFuncs->findModuleForPackage(currentThread, thisVMClassLoader, packageName, (U_32)packageLength);
```

Comment:
The APIs seem orthogonal to me. For example why does `isAnyClassLoadedFromPackage` accept a `U_8*` and `UDATA` to represent a UTF8 string and a length and `findModuleForPackageUTF8` needs a `J9UTF8*`? Why aren't they all synchronized on one set of APIs. Otherwise we get into these situations where we need to mix and match.

Reply:
Fixed in d05ffd5

Reply:
`J9UTF8` structure is a `U_8*` string (without `null` terminator) and a length at beginning of the structure. `J9UTF8` requires `J9UTF8_DATA`, `J9UTF8_SET_LENGTH` and `J9UTF8_LENGTH` for accessing its fields. There are cases that the APIs expect the data and length without going through the getter/setter. In all these scenarios, the data are not ended with `null` and the end are determined by `LENGTH`. 
Hope this makes sense.